tests/backup/backup_helper.go

```diff
@@ -1254,6 +1292,213 @@ func IsBackupLocationPresent(bkpLocation string, ctx context.Context, orgID stri
+func GetOrdinalScheduleBackupNameAfterCheck(ctx context.Context, scheduleName string, ordinal int, orgID string) (string, error) {
+	ordinalScheduleBackupName, err := task.DoRetryWithTimeout(func() (interface{}, bool, error) {
+		ordinalScheduleBackupName, err := GetOrdinalScheduleBackupName(ctx, scheduleName, ordinal, orgID)
+		if err != nil {
+			if ordinal < 1 {
+				return "", false, err
+			}
+			return "", true, err
+		}
+		return ordinalScheduleBackupName, false, nil
+	}, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second)
+	if err != nil {
+		return "", err
+	}
+	_, err = backupSuccessCheck(ordinalScheduleBackupName.(string), orgID, 0, 0, ctx)
+	if err != nil {
+		return "", err
+	}
+	return ordinalScheduleBackupName.(string), nil
+}
+
+// GetFirstScheduleBackupName returns the name of the first schedule backup for the given schedule
+func GetFirstScheduleBackupName(ctx context.Context, scheduleName string, orgID string) (string, error) {
+	allScheduleBackupNames, err := Inst().Backup.GetAllScheduleBackupNames(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	if len(allScheduleBackupNames) == 0 {
+		return "", fmt.Errorf("no backups found for schedule %s", scheduleName)
+	}
+	return allScheduleBackupNames[0], nil
+}
+
+// GetFirstScheduleBackupNameAfterCheck returns the name of the first schedule backup for the given schedule after success check
+func GetFirstScheduleBackupNameAfterCheck(ctx context.Context, scheduleName string, orgID string) (string, error) {
+	firstScheduleBackupName, err := GetFirstScheduleBackupName(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	_, err = backupSuccessCheck(firstScheduleBackupName, orgID, 0, 0, ctx)
+	if err != nil {
+		return "", err
+	}
+	return firstScheduleBackupName, nil
+}
+
+// GetLatestScheduleBackupName returns the name of the latest schedule backup for the given schedule
+func GetLatestScheduleBackupName(ctx context.Context, scheduleName string, orgID string) (string, error) {
+	allScheduleBackupNames, err := Inst().Backup.GetAllScheduleBackupNames(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	if len(allScheduleBackupNames) == 0 {
+		return "", fmt.Errorf("no backups found for schedule %s", scheduleName)
+	}
+	return allScheduleBackupNames[len(allScheduleBackupNames)-1], nil
+}
+
+// GetLatestScheduleBackupNameAfterCheck returns the name of the latest schedule backup for the given schedule after success check
+func GetLatestScheduleBackupNameAfterCheck(ctx context.Context, scheduleName string, orgID string) (string, error) {
+	latestScheduleBackupName, err := GetLatestScheduleBackupName(ctx, scheduleName, orgID)
+	if err != nil {
+		return "", err
+	}
+	_, err = backupSuccessCheck(latestScheduleBackupName, orgID, 0, 0, ctx)
+	if err != nil {
+		return "", err
+	}
+	return latestScheduleBackupName, nil
+}
```

Comment:
Same here. Not required.

Reply:
The use case is similar to GetOrdinalScheduleBackupNameAfterCheck but in this case we will never know the ordinal of latest schedule backup precisely beforehand.

Reply:
Removed GetLatestScheduleBackupNameAfterCheck.